bindings/pyroot/pythonizations/python/ROOT/_pythonization/_tgraph.py

```diff
@@ -34,9 +37,51 @@ def set_size(self, buf):
 # The pythonization consists in setting the size of the array that the getter
 # method returns, so that it is known in Python and the array is fully usable
 # (its length can be obtained, it is iterable).
-comp = cppyy.py.compose_method('^TGraph(2D)?$|^TGraph.*Errors$', # class to match
-                               'GetE?[XYZ]$',                    # method to match
-                               set_size)                         # post-process function
+comp = cppyy.py.compose_method(
+    "^TGraph(2D)?$|^TGraph.*Errors$", "GetE?[XYZ]$", set_size  # class to match  # method to match
+)  # post-process function
 
 # Add the composite to the list of pythonizors
 cppyy.py.add_pythonization(comp)
+
+
+def _init(self, *args):
+    import numpy as np
+
+    if isinstance(args[0], (np.ndarray, np.generic)):
+        self.__class__.__init__(args[0].size, *args)
+
+
+def GetAsNumpy(self):
+    # Helper to create a numpy array from a raw array pointer.
+    #
+    # Note: The data is copied.
+    #
+    # Args:
+    #     buffer (cppyy.LowLevelView):
+    #         The pointer to the beginning of the array data, usually
+    #         obtained from a C++ function that returns a `double *`.
+    #
+    # Returns:
+    #     numpy.ndarray
+    import numpy as np
+
+    # Check if buffer is a nullptr by using implicit conversion from
+    # `nullptr` to Bool (for some reason comparing with ROOT.nullptr
+    # doesn't work).
+    a, b = np.copy(np.frombuffer(self.Get(), dtype=np.float64, count=len(self)))
+    return a, b
+
+
+python_funcs = [GetAsNumpy]
+
+
+@pythonization("TGraph")
+def pythonize_tgraph(klass):
+    # Parameters:
+    # klass: class to be pythonized
+    # Support hist *= scalar
+    klass.__init__ = _init
```

Comment:
Here you are losing the original implementation of `__init__` provided by cppyy, I'd save it in another attribute so it can be used later.

Reply:
See for example the `_drawables.py` pythonization file.